shared-services/src/files/file-service.provider.ts

```diff
@@ -4,9 +4,10 @@ import { FileServiceInterface } from "./file-service.interface"
 import { FileConfig, FileServiceTypeEnum } from "./file-config"
 
 export class FileServiceProvider {
-  service: FileServiceInterface
+  static publicUploadService: FileServiceInterface
+  static privateUploadService: FileServiceInterface
 
-  private configure(): FileConfig {
+  private static configure(): FileConfig {
```

Comment:
This should _take_ a config object rather than generating one.  The config should be validated (and throw an exception if anything is missing/wrong or if it has already been called), then stored into a static var to be used on line 22 below rather than calling `configure` again.  This gives each calling service control over how to handle the configuration and avoids hiding env vars in external libraries.  Just export the config types (see above) so that they can be used directly.

Note too that the input to the `configure` method will need to have the ability to define both public and private services at the minimum.  If we want to be able to define additional legacy services (which would make transition between services easier), then we would also need a `Record<string, IndividualServiceConfig>` (or `Array<IndividualServiceConfig>` if the config has a name value).  These would likely be hardcoded into the config input in practice since they would be unlikely to ever change at that point.  

An example `FileProviderConfig` type might look something like this:

```
type FileProviderConfig = {
  publicService: FileServiceConfig
  privateService: FileServiceConfig
  additionalServices: Array<FileServiceConfig>
  // OR
  additionalServices: Record<string, FileServiceConfig>
}
```

We would then configure the provider like so:

```
FileServiceProvider.configure({
  publicService: <publicServiceConfig>,
  privateService: <privateServiceConfig>,
  additionalServices: {
    cloudinary: <legacyCloudinaryConfig>
  }
  // OR
  additionalServices: [
    <legacyCloudinaryConfig>
  ]
})
```

Reply:
I add the `additionalServices` part above to provide a better idea on a good direction to head in, but I don't think that needs to be added at this point.  I do think that providing a means to name services ("s3-public", "cloudinary-exygy", etc) will ultimately make the library more robust and future-resistant, but I don't think it's necessary to add in this PR.

Reply:
Thanks for the suggestions, I made the changes to pass in the config object and have it be validated in the configure() method. However this breaks the public site, since it fails on the check that the cloud upload preset should be specified. The public site only downloads files so it doesn't call putFile() and doesn't need the upload preset, which is why it's not part of the .env file. Should I add it to the .env file even if it is not required? Or is there a better place than the public and partner _app.tsx to call FileServiceProvider.configure()?

I had also initially added the check to throw an exception if configure() is called when the services are already configured and that exception was thrown when I tried to bring up the partners site, so I removed that check for now. I imagine there could be a better way to call configure() so it's only called once.

Reply:
Never mind, I misunderstood about the upload preset parameter that is specific to Cloudinary.  It looks like that's optional for signed uploading, so I think it's safe to treat it as an optional config value as well.

Reply:
I removed the validation for the upload preset env variable, and added a check to not configure the services again if they are already defined. Both the public and the partners sites come up now and listing photo uploading/downloading works on the partners site.